### ISSUE TITLE
id field in the example

### ISSUE THREAD

**Issue by imran3180** [pytorch/text#625]
## ❓ Questions and Help

**Description**

How can I find the id of the particular example while processing the batch?

I need it because I have to maintain a particular statistics for each example while training. Since Iterators shuffle the example while batching, so the order is getting lost. 

Or Is there any workaround by which I can identify the reverse mapping of examples in batch to the original datasets.?

**Comment by zhangguanheng66** [comment]
a particular statistics? So you mean a non-uniform distribution? I guess you probably need a custom `Sampler` and batch examples based a sampling density.

**Comment by imran3180** [comment]
I was asking a way by which we can access id(the unique identifier of the example after numericalzation) in the training loop. 
using torchtext.datasets.MultiNLI does not return pairID of the example in the batch. 

So I used the workaround by using fields dictionary and TabularDataset. 

```python
fields = {
     'sentence1': ('premise', self.TEXT),
     'sentence2': ('hypothesis', self.TEXT),
     'gold_label': ('label', self.LABEL),
     'pairID': ('id', self.ID)
}
```
and then use the data.TabularDataset.

Closing this issue.
